test/functional/mempool_reorg.py

```diff
@@ -106,8 +106,45 @@ def run_test(self):
 
         self.log.info("Check that the mempool is empty")
         assert_equal(set(self.nodes[0].getrawmempool()), set())
+        assert_equal(set(self.nodes[1].getrawmempool()), set())
         self.sync_all()
 
 
+        self.log.info("Simulate natural reorg")
+        self.connect_nodes(0, 1)
+        assert_equal(self.nodes[0].getblockcount(), self.nodes[1].getblockcount())
+        assert_equal(set(self.nodes[0].getrawmempool()), set(self.nodes[0].getrawmempool()))
+        self.generate(self.nodes[0], 3)
+        # Create another transaction which is time-locked to two blocks in the future
+        utxo = wallet.get_utxo()
+        timelock_tx_2 = wallet.create_self_transfer(
+            from_node=self.nodes[0],
+            utxo_to_spend=utxo,
+            mempool_valid=False,
+            locktime=self.nodes[0].getblockcount() + 2
+        )['hex']
+        # Check that the time-locked transaction is too immature to spend
+        assert_raises_rpc_error(-26, "non-final", self.nodes[0].sendrawtransaction, timelock_tx_2)
+        assert_raises_rpc_error(-26, "non-final", self.nodes[1].sendrawtransaction, timelock_tx_2)
+
+        self.log.info("Disconnect nodes and mine separate forks")
+        self.disconnect_nodes(0, 1)
+        # Don't try to sync nodes because they are disconnected
+        no_sync_func = lambda: True
```

Comment:
nit: Why is this needed? You can use `self.no_op`

Reply:
ahh, didn't know about that - fixed